src/wallet/rpcwallet.cpp

```diff
@@ -158,6 +158,30 @@ static std::string LabelFromValue(const UniValue& value)
     return label;
 }
 
+static void SetFeeEstimateMode(CWallet* const pwallet, CCoinControl& cc, const UniValue& estimate_mode, const UniValue& estimate_param, bool may_set_rbf = true)
+{
+    if (!estimate_mode.isNull()) {
+        if (!FeeModeFromString(estimate_mode.get_str(), cc.m_fee_mode)) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid estimate_mode parameter");
+        }
+    }
+
+    if (estimate_param.isNull()) return;
+
+    if (cc.m_fee_mode == FeeEstimateMode::BTC_KB || cc.m_fee_mode == FeeEstimateMode::SAT_B) {
+        CAmount fee_rate = AmountFromValue(estimate_param);
+
+        if (cc.m_fee_mode == FeeEstimateMode::SAT_B) {
+            fee_rate /= 100000; // 1 sat / B = 0.00001 BTC / kB
+        }
+        cc.m_feerate = CFeeRate(fee_rate);
+        // default RBF to true for explicit fee rate mode
+        cc.m_signal_bip125_rbf = (cc.m_signal_bip125_rbf ? *cc.m_signal_bip125_rbf : false) || may_set_rbf;
```

Comment:
This seems unnecessary (or you need another test case). You can drop `may_set_rbf` and use `if(cc.m_signal_bip125_rbf) {*cc.m_signal_bip125_rbf = true}`. See 4594923046585453beef83e833e6a422a78b66b2

Reply:
Shouldn't that be `if (!cc.m_signal_bip125_rbf)`? I thought optionals returned `0` when unset, and we only wanna set it if it's not been set already, right?

Reply:
Wading through the slugfest that is boost, it seems that the most correct way to determine if an optional is uninitialized is to do `== boost::none`. I.e.
```C++
if (cc.m_signal_bip125_rbf == boost::none) { ... }
```

Reply:
+1 for using `boost::none`